monai/networks/nets/vista3d.py

```diff
@@ -0,0 +1,957 @@
+        point_head: point head used for interactive segmetnation
+    """
+    def __init__(self, image_encoder, class_head, point_head):
+        super().__init__()
+        self.image_encoder = image_encoder
+        self.class_head = class_head
+        self.point_head = point_head
+        self.image_embeddings = None
+        self.auto_freeze = False
+        self.point_freeze = False
+        self.NINF_VALUE = -9999
+        self.PINF_VALUE = 9999
+
+    def get_bs(self, class_vector, point_coords):
+        """ Get number of foreground classes based on class and point prompt.
+        """
+        if class_vector is None:
+            assert point_coords is not None, "prompt is required"
+            return point_coords.shape[0]
+        else:
+            return class_vector.shape[0]
+
+    def convert_point_label(self, point_label, label_set=None,
+                            special_index=[23, 24, 25, 26, 27, 57, 128]):
+        """ Convert point label based on its class prompt. For special classes defined in special index,
+        the positive/negative point label will be converted from 1/0 to 3/2. The purpose is to separate those
+        classes with ambiguous classes.
+        """
+        if label_set is None:
+            return point_label
+        assert point_label.shape[0] == len(label_set)
+        for i in range(len(label_set)):
+            if label_set[i] in special_index:
+                for j in range(len(point_label[i])):
+                    point_label[i, j] = (
+                        point_label[i, j] + 2
+                        if point_label[i, j] > -1
+                        else point_label[i, j]
+                    )
+        return point_label
+
+    def sample_points_patch_val(
+        self,
+        labels,
+        patch_coords,
+        label_set,
+        use_center=True,
+        mapped_label_set=None,
+        max_ppoint=1,
+        max_npoint=0,
+        **kwargs
+    ):
+        """Sample points for patch during sliding window validation. Only used for point only validation.
+        Args:
+            labels: [1, 1, H, W, D]
+            patch_coords: sliding window slice object
+            label_set: local index, must match values in labels
+            use_center: sample points from the center
+            mapped_label_set: global index, it is used to identify special classes.
+            max_ppoint/max_npoint: positive points and negative points to sample.
+        """
+        point_coords, point_labels = sample_points_from_label(
+            labels[patch_coords],
+            label_set,
+            max_ppoint=max_ppoint,
+            max_npoint=max_npoint,
+            device=labels.device,
+            use_center=use_center,
+        )
+        point_labels = self.convert_point_label(point_labels, mapped_label_set)
+        return (
+            point_coords,
+            point_labels,
+            torch.tensor(mapped_label_set).to(point_coords.device).unsqueeze(-1),
```

Comment:
HI @heyufan1995 , I think this place is wrong, we should use label_set

Reply:
this place is mapped label set. I added a docstring.